File: full_case_study_solns/_d_mass/python/hw7/massParamHW7.py
# mass Parameter File
import numpy as np
# import control as cnt
import sys
sys.path.append('..')  # add parent directory
import massParam as P

# PD gains
#kp = 4.5
#kd = 12.0

# open loop coefficients
a1 = P.b/P.m
a0 = P.k/P.m

# select desired closed loop char eq
Delta_cl_d = np.poly([-1,-1.5])

# PD gains
kp = P.m*(Delta_cl_d[2]-a0)
kd = P.m*(Delta_cl_d[1]-a1)


print('kp: ', kp)
print('kd: ', kd)

File: full_case_study_solns/_d_mass/matlab_old/hw7/massParamHW7.m
addpath ./..
massParam

Delta_ol = [1,P.b/P.m,P.k/P.m];
p_ol = roots(Delta_ol);

Delta_cl_d = poly([-1,-1.5]);

P.kp = P.m*(Delta_cl_d(3)-Delta_ol(3));
P.kd = P.m*(Delta_cl_d(2)-Delta_ol(2));
